cmake/platform/windows.cmake

```diff
@@ -88,3 +88,28 @@ if(MSVC)
   # https://blogs.msdn.microsoft.com/vcblog/2018/04/09/msvc-now-correctly-reports-__cplusplus/
   add_compile_options(/Zc:__cplusplus)
 endif()
+
+#
+# Helper macros added to the catkin system to replace Windows lack of shebang support.
+#
+# Code courtesy of Yujin Robot.  Derived from https://github.com/ros-windows/win_ros
+#
+
+# distutils does not have support for 'entry_points'
+macro(add_windows_python_batch_helper name)
+  set(PYTHON_TARGET ${name})
+  configure_file(${catkin_EXTRAS_DIR}/templates/python_wrapper.bat.in
+                 ${CATKIN_DEVEL_PREFIX}/bin/${name}.bat)
+  install(PROGRAMS ${CATKIN_DEVEL_PREFIX}/bin/${PYTHON_TARGET}.bat 
+          DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION})
+endmacro()
+
+# some python scripts need an executable, not a batch file
+macro(add_windows_python_exe_helper name)
```

Comment:
This should be a function (since it doesn't need to be a macro).

Reply:
Maybe perform the platform check within the function so it can be called unconditionally?

Reply:
change to function and move platform check within the function